lib/active_merchant/billing/gateways/cyber_source.rb

```diff
@@ -539,6 +541,16 @@ def add_address(xml, payment_method, address, options, shipTo = false)
         end
       end
 
+      def address_names(address_name, payment_method)
+        names = split_names(address_name)
+        return names if names.any?(&:present?)
```

Comment:
❤️ My notes: this makes sure that the first and last name come from the same source, either the provided `address_name` or  `payment_method`.